### PR TITLE
AX-3866: Poynt collect enhancements - charge using nonce

### PR DESCRIPTION
**Pull request by ivo-siddarta**

### CODE REVIEW

File: lib/collect.js [hunk 1]
```diff
@@ -71,11 +73,39 @@ module.exports.chargeToken = function (options, next) {
...
+      name: 'InvalidArgumentError',
+      message: 'Invalid params: Request should contain either nonce or token',
+      statusCode: 400
+    });
+    return next(hasErr);
+  }
+
+  // AX-3758: charge using nonce as funding source
+  if (options.nonce) {
+    var hasErr = helpers.hasKeys(options, ["tokenize"]);
```
[the remark this answers is not in the document]
Comment (sandipPoynt) [reply lib/collect.js:90]: yes its optional

File: lib/collect.js [hunk 2]
```diff
@@ -71,11 +73,39 @@ module.exports.chargeToken = function (options, next) {
...
+    if (hasErr) {
+      return next(hasErr);
+    }
+    body.fundingSource = {
+      nonce: options.nonce,
```
Comment (sandipPoynt) [inline lib/collect.js:91]: it should be either of them, does this one mean both ?

Comment (sandipPoynt) [reply lib/collect.js:91]: we cant have both, it will be confusing.

Comment (ivo-siddarta) [reply lib/collect.js:91]: either

Comment (ivo-siddarta) [reply lib/collect.js:91]: line 79 does the checking if there exist both

**Review by vchau — APPROVED** [review]

**Pull request closed without merging**